assets/js/components/data/index.js

```diff
@@ -196,33 +196,35 @@ const dataAPI = {
 			data: { request: currentRequest },
 			method: 'POST',
 		} ).then( ( results ) => {
-			each( results, ( result, key ) => {
-				if ( ! keyIndexesMap[ key ] ) {
-					console.debug( 'data_error', 'unknown response key ' + key ); // eslint-disable-line no-console
+			Object.entries( results ).forEach( ( [ requestKey, response ] ) => {
+				if ( ! keyIndexesMap[ requestKey ] ) {
+					console.debug( 'data_error', 'unknown response key ' + requestKey ); // eslint-disable-line no-console
 					return;
 				}
 
-				const isError = isWPError( result );
-				if ( isError ) {
-					const { datapoint, type, identifier } = dataRequest[ keyIndexesMap[ key ] ];
```

Comment:
This seems to have been possible due to this odd behavior when using an array as a key:

![image](https://user-images.githubusercontent.com/1621608/104291772-a09e8a80-54c4-11eb-8f3f-13855e298cc4.png)

This allowed it to work for single requests, but it would fail as soon as there were multiple requests for data with the same cache key, in the same `combinedRequest`. This only affected requests that resulted in an error, but still not good of course.